product.py

```diff
@@ -5,3 +5,191 @@
     :copyright: (c) 2015 by Openlabs Technologies & Consulting (P) Limited
     :license: BSD, see LICENSE for more details.
 """
+from trytond.model import ModelSQL, ModelView, fields
+from trytond.pool import PoolMeta
+from trytond.pyson import Eval
+from trytond.transaction import Transaction
+
+__metaclass__ = PoolMeta
+__all__ = [
+    'ProductAttributeSet', 'ProductAttributeSelectionOption',
+    'ProductAttribute', 'ProductAttributeAttributeSet',
+    'Template', 'ProductProductAttribute', 'Product',
+]
+
+ATTRIBUTE_TYPES = [
+    ('boolean', 'Boolean'),
+    ('integer', 'Integer'),
+    ('char', 'Char'),
+    ('float', 'Float'),
+    ('numeric', 'Numeric'),
+    ('date', 'Date'),
+    ('datetime', 'DateTime'),
+    ('selection', 'Selection'),
+]
+
+
+class ProductAttributeSet(ModelSQL, ModelView):
+    "Product Attribute Set"
+    __name__ = 'product.attribute.set'
+
+    name = fields.Char('Name', required=True, translate=True)
+    attributes = fields.Many2Many(
+        'product.attribute-product.attribute-set',
+        'attribute_set', 'attribute', 'Attributes'
+    )
+
+
+class ProductAttributeSelectionOption(ModelSQL, ModelView):
+    "Attribute Selection Option"
+
+    __name__ = 'product.attribute.selection_option'
+
+    name = fields.Char("Name", required=True)
+    attribute = fields.Many2One("product.attribute", "Attribute", required=True)
+
+
+class ProductAttribute(ModelSQL, ModelView):
+    "Product Attribute"
+    __name__ = 'product.attribute'
+
+    sets = fields.Many2Many(
+        'product.attribute-product.attribute-set',
+        'attribute', 'attribute_set', 'Sets'
+    )
+
+    name = fields.Char('Name', required=True)
+    string = fields.Char('String', translate=True, required=True)
```

Comment:
I will call this "Display Name" and not make it required

The rec_name name should then show ``self.display_name or self.name``